### PR TITLE
refactor: move CTAS/CVAS field II

### PR DESCRIPTION
**Pull request by hughhhh**
### SUMMARY
Moved CTAS/CVAS Schema Field Closer to the "Allow Create Table As" In Edit Datasource Modal. This required nested conditional rendering, so I had to use some useState/useEffect in order to hold that state at a higher level for all functions to run properly.

#### Field before:
ctascvas-before
#### Field after:
ctascvas-after-1
ctascvas-after-2
ctascvas-after-3

### TEST PLAN
Tests will ensure that all elements are rendered in their respectively appropriate conditions.

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by AAfghahi — APPROVED** [review]

File: superset-frontend/spec/javascripts/sqllab/fixtures.ts [hunk 1]
```diff
@@ -480,6 +480,8 @@ export const initialState = {
       DEFAULT_SQLLAB_LIMIT: 1000,
       SQL_MAX_ROW: 100000,
       DISPLAY_MAX_ROW: 100,
+      SQLALCHEMY_DOCS_URL: 'test_SQLALCHEMY_DOCS_URL',
+      SQLALCHEMY_DISPLAY_TEXT: 'test_SQLALCHEMY_DISPLAY_TEXT',
```
Comment (eschutho) [inline superset-frontend/spec/javascripts/sqllab/fixtures.ts:484]: wasn't this part of a previous PR?

File: superset-frontend/spec/javascripts/views/CRUD/data/database/DatabaseModal_spec.jsx [hunk 2]
```diff
@@ -19,71 +19,243 @@
...
-    expect(wrapper.find(Tabs.TabPane)).toExist();
-    expect(wrapper.find(Tabs.TabPane)).toHaveLength(5);
-  });
+      // While 'Expose in SQL Lab' is checked, all settings should display
+      expect(exposeInSqlLab).not.toBeChecked();
+
+      // When clicked, "Expose in SQL Lab" becomes unchecked
+      userEvent.click(exposeInSqlLab);
+
+      // While unchecked, only "Expose in SQL Lab" should display
```
Comment (eschutho) [inline superset-frontend/spec/javascripts/views/CRUD/data/database/DatabaseModal_spec.jsx:158]: Are the comments still aligned with the code? This says unchecked, while the assertion below is checked.

File: superset-frontend/spec/javascripts/views/CRUD/data/database/DatabaseModal_spec.jsx [hunk 3]
```diff
@@ -19,71 +19,243 @@
...
-    expect(wrapper.find('input[name="database_name"]')).toExist();
-    expect(wrapper.find('input[name="sqlalchemy_uri"]')).toExist();
+      // Check both "Allow CTAS" and "Allow CVAS" to reveal schema field
+      userEvent.click(allowCTAS);
+      userEvent.click(allowCVAS);
+      expect(schemaField).toHaveClass('open');
+      // Uncheck both "Allow CTAS" and "Allow CVAS" to hide schema field again
+      userEvent.click(allowCTAS);
+      // Both checkboxes go unchecked, so the field should no longer render
+      userEvent.click(allowCVAS);
```
Comment (eschutho) [inline superset-frontend/spec/javascripts/views/CRUD/data/database/DatabaseModal_spec.jsx:257]: nit, but I would move this up a line with the other click to be more in line with the comments.

**Comment by eschutho** [comment]
We changed some functionality of the indeterminate checkbox, so we'll need to test across the site cc @rusackas @junlincc
I'm doing some visual testing now.

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal.tsx [hunk 4]
```diff
@@ -43,6 +43,9 @@ interface DatabaseModalProps {
 }
 
 const DEFAULT_TAB_KEY = '1';
+const EXPOSE_SQLLAB_FORM_HEIGHT = '260px';
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal.tsx:46]: can you bump this up to 270px? It's cutting some off. This approach is admittedly brittle. Maybe we can try to find a way of capturing the current height of the children and adding it up and passing it to emotion for styling in the next iteration of this.

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal.tsx [hunk 5]
```diff
@@ -587,7 +639,7 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
               <IndeterminateCheckbox
                 id="impersonate_user"
                 indeterminate={false}
-                checked={db ? !!db.impersonate_user : false}
+                checked={!!db?.impersonate_user}
                 onChange={onInputChange}
               />
               <div>{t('Impersonate Logged In User (Presto & Hive)')}</div>
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal.tsx:645]: you may want to move all of these into `labelText` as well.

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal.tsx [hunk 6]
```diff
@@ -69,11 +83,15 @@ const StyledInputContainer = styled.div`
 
   .input-container {
     display: flex;
-    align-items: center;
+    align-items: top;
 
     label {
       display: flex;
       margin-right: ${({ theme }) => theme.gridUnit * 2}px;
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal.tsx:90]: I think you'll want to remove this as well, but double check when you have the other labels moved into the correct label tag.

**Comment by eschutho** [comment]
Uploading Screen Recording 2021-03-31.mov…

Other checkboxes:
existing:
Screenshot_3_31_21__3_57_PM
new:
Screenshot_3_31_21__3_57_PM
existing:
_DEV__Superset
new:
Superset

**Comment by eschutho** [comment]
@hughhhh @lyndsiWilliams indeterminate checkboxes are only on this modal and the list views. The list views looked ok. I would just make sure that all of the checkboxes in the modal are passing in the label and then just do a check on updating any styles.

File: superset-frontend/spec/javascripts/sqllab/fixtures.ts (hunk 1 above)
Comment (hughhhh) [reply superset-frontend/spec/javascripts/sqllab/fixtures.ts:484]: not sure but we wanted to update the fixture to make sure it has all the config items

**Comment by eschutho** [comment]
looks great! I rebased this recently and saw that an earlier commit had a few trims on database_name in the DatabaseModal file. Do we need to add those in? 
`db.database_name.trim()`

**Comment by eschutho** [comment]
tested in ephemeral environment, too, and it looks good. I wasn't able to test save functionality b/c of sqllite but I had tested it locally previously.

**Comment by hughhhh** [comment]
> looks great! I rebased this recently and saw that an earlier commit had a few trims on database_name in the DatabaseModal file. Do we need to add those in?
db.database_name.trim()
Here's the PR if you want to make sure that they changes are represented here: #13800 (files)

Those changes we need are currently on the branch already, the other `database_name` reference don't need to have `trim`. We can add them if want the consistency

**Review by betodealmeida — APPROVED** [review]
Nice!

**Pull request merged**